.github/autobuild/mac.sh

```diff
@@ -73,6 +73,9 @@ pass_artifact_to_job() {
 case "${1:-}" in
     setup)
         setup
+        # set up the macos_signed output if needed, but prevent
+        # a return status of 1 from propagating to the script exit status.
+        prepare_signing || true
```

Comment:
Hmm, so we would now be running those steps twice?
Maybe there's some nice way to save that state and skip the second run somehow. Either pass back toe macos_signed output variable as an environment variable or touch some kind of file.

Reply:
Well in fact, the only parts of `prepare_signing` that are needed in the setup phase are the initial tests and the setting of the output variable (so that it can be tested in the workflow). So we could make a different function `check_signing` that just does those parts, and maybe even call it from `prepare_signing` to reactor those parts out.